src/Processors/Formats/Impl/ParquetBlockOutputFormat.cpp

```diff
@@ -328,6 +329,7 @@ void ParquetBlockOutputFormat::writeUsingArrow(std::vector<Chunk> chunks)
         parquet::WriterProperties::Builder builder;
         builder.version(getParquetVersion(format_settings));
         builder.compression(getParquetCompression(format_settings.parquet.output_compression_method));
+        builder.compression_level(static_cast<int>(format_settings.parquet.output_compression_level));
```

Comment:
@arthurpassos This is an issue, since snappy codec doest't support setting compression level. If so arrow will retutrn "Invalid: Codec 'snappy' doesn't support setting a compression level."

Reply:
hm.. It is funny that no tests caught it, and doing a grep it seems like there are a few tests for parquet output.


I'll investigate

Reply:
I also see the below on [docs](https://github.com/ClickHouse/arrow/blob/d3a576d03ad943773d9c55974f9c797a5042b4a8/cpp/src/parquet/properties.h#L431-L432):

> If the compressor does not allow for selecting different
    compression levels, calling this function would not have any effect.

Reply:
Ok, it seems like it really happens if we set `output_format_parquet_use_custom_encoder=0 and output_format_parquet_compression_method='snappy'`